Source/JavaScriptCore/wasm/WasmMemory.cpp

```diff
@@ -386,12 +386,15 @@ RefPtr<Memory> Memory::tryCreate(VM& vm, PageCount initial, PageCount maximum, M
     ASSERT(initial);
     RELEASE_ASSERT(!maximum || maximum >= initial); // This should be guaranteed by our caller.
 
-    const size_t initialBytes = initial.bytes();
-    const size_t maximumBytes = maximum ? maximum.bytes() : 0;
+    const uint64_t initialBytes = initial.bytes();
+    const uint64_t maximumBytes = maximum ? maximum.bytes() : 0;
 
     if (initialBytes > MAX_ARRAY_BUFFER_SIZE)
         return nullptr; // Client will throw OOMError.
 
+    if (maximumBytes > MAX_ARRAY_BUFFER_SIZE)
```

Comment:
Can you ensure the other 64bit engines' behavior on this? If maximum does not have a limit on 64bit environment, then we should not introduce this check for 64bit build.

Reply:
fixed